app/javascript/mastodon/features/community_timeline/index.js

```diff
@@ -89,27 +92,17 @@ export default class CommunityTimeline extends React.PureComponent {
     dispatch(expandCommunityTimeline({ maxId, onlyMedia }));
   }
 
-  handleHeadlineLinkClick = e => {
-    const { columnId, dispatch } = this.props;
-    const onlyMedia = /\/media$/.test(e.currentTarget.href);
-
-    dispatch(changeColumnParams(columnId, { other: { onlyMedia } }));
+  handleSettingChanged = (key, checked) => {
+    const { columnId } = this.props;
+    if (!columnId && key[0] === 'other' && key[1] === 'onlyMedia') {
+      this.context.router.history.replace(`/timelines/public/local${checked ? '/media' : ''}`);
+    }
```

Comment:
Actually, on thinking more about it… is the current state of the PR that Redux is being used for the toggle but the URL is being used for the filtering?? Because you should adjust it so that Redux is being used for everything if so. (Just by inserting `['params', 'other', 'onlyMedia']` into the props at the appropriate places and then reading from that instead of the URL.)

Reply:
Pressing the back button doesn't undo the toggle, since I'm using .replace instead of .push

However, you're right that URL state isn't a terribly elegant way to do this. When I was writing this my goal was to reach parity with the tab-based version, which meant doing things the same way it did them.

Also, at the time, I wasn't sure how to change a parameter of the mounted column, as it doesn't have a column id (so changeColumnParams wouldn't work). I think I have a better understanding of the system now though, so I'll see what I can do.